main.py

```
@@ -41,6 +43,40 @@ def __init__(self, owner: str, repo: str, plugins_dir: Path):
+            return ''
+
+        # On Flatpak, /app is read-only at runtime; pip installs will always fail.
+        if os.getenv('FLATPAK_ID'):
+            return (
+                f"\n\nDependency installation skipped: running inside Flatpak "
+                f"(read-only filesystem).\n"
+                f"Install manually: pip install -r \"{req_file}\""
+            )
+
+        try:
+            result = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)